File: PATTERNS/Pattern31/Pattern31.c
#include<stdio.h>

void main(){
	int i,j,n;
	printf("enter the number:");
	scanf("%d",&n);
	n=n*2-1;
	for(i=1;i<=n;i++){
		for(j=1;j<=n;j++){
			if(i==j||j==(n+1)-i){
				printf("*");
			}
			else{
				printf(" ");
			}
		}
		printf("\n");
	}
}

File: PATTERNS/Pattern31/Pattern31.java

/*
 *       * 
  *     *  
   *   *   
    * *
     *   
    * *    
   *   *   
  *     *  
 *       * 
 
 */
import java.util.*;

public class Pattern31 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the row number: ");
        int n = input.nextInt();

        for (int i = 1; i < 2 * n; i++) {
            int flag, flag1 = n;
            if (i <= n) {
                flag = i;
            } else {
                flag = 2 * n - i;
            }
            for (int k = 1; k <= flag; k++) {
                System.out.print(" ");
            }
            for (int j = flag1; j >= flag; j--) {
                if (j == flag1 || j == flag) {
                    System.out.print("* ");
                } else {
                    System.out.print("  ");
                }
            }
            System.out.println();
        }
    }
}